xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-ui/src/main/resources/XWiki/DeletedDocuments.xml

```diff
@@ -349,7 +390,32 @@ document.observe('xwiki:dom:loaded', function() {
   if ($('documentsTrash')) {
     $('documentsTrash').up('.hidden').removeClassName('hidden');
   }
-});</code>
+});
+
+// we need to wait the complete display of the livetable to get the number of rows
+document.observe('xwiki:livetable:displayComplete', function() {
+  if ($('documentsTrash')) {
+    var actionAllContainer = $('documentsTrash').getElementsBySelector('td.xwiki-livetable-display-header-filter:last-child')[0];
+    var trashAllDisabled = (trashDocumentsLivetable.totalRows > 0) ? false : true;
+
+    // avoid to get multiple times the button
+    if ($('trashAllDeleted')) {
+      $('trashAllDeleted').remove();
+    }
+
+    var trashAll = new Element('a', {
+        'id': 'trashAllDeleted',
+        'class':'btn btn-danger',
+        'href':'#',
+        'disabled': trashAllDisabled
```

Comment:
Shouldn't the button (link) appear as disabled? I would use a CSS class instead. See https://getbootstrap.com/docs/3.3/css/#anchor-element-1 .

Reply:
Both work, but I think the CSS class is more easy to understand in this particular case because you're using a link. At least I didn't know why it works with the attribute until I checked the Bootstrap styles. While when you see the CSS class you assume there is some CSS to make it look disabled. Just my opinion.